ui/lib/core/addon/components/alert-banner.js

```diff
@@ -13,36 +11,14 @@ import layout from '../templates/components/alert-banner';
  * ```
  *
  * @param {String} type=null  - The banner type. This comes from the message-types helper.
- * @param {String} [secondIconType=null] - If you want a second icon to appear to the right of the title. This comes from the message-types helper.
- * @param {Object} [progressBar=null] - An object containing a value and maximum for a progress bar. Will be displayed next to the message title.
  * @param {String} [message=null] - The message to display within the banner.
+ * @param {Object} [progressBar=null] - An object containing a value and maximum for a progress bar. Will be displayed next to the message title.
+ * @param {Boolean} [showLoading=false] - Shows a loading icon to the right of the title.
  * @param {String} [title=null] - A title to show above the message. If this is not provided, there are default values for each type of alert.
- * @param {String} [bannerType=alert] - Defaults to 'alert', can be used to specify an alert banner's test selector
- *
  */
 
-export default Component.extend({
-  layout,
-  type: null,
-  message: null,
-  title: null,
-  secondIconType: null,
-  progressBar: null,
-  yieldWithoutColumn: false,
-  marginless: false,
-  classNameBindings: ['containerClass'],
-  bannerType: 'alert',
-
-  containerClass: computed('type', 'marginless', function () {
-    const base = this.marginless ? 'message message-marginless ' : 'message ';
-    return base + messageTypes([this.type]).class;
-  }),
-
-  alertType: computed('type', function () {
-    return messageTypes([this.type]);
-  }),
-
-  secondAlertType: computed('secondIconType', function () {
-    return messageTypes([this.secondIconType]);
-  }),
-});
+export default class AlertBanner extends Component {
+  get alertType() {
+    return messageTypes([this.args.type]);
```

Comment:
Nice to have: validation in the form of an `assert` that the `@type` is a valid option

Reply:
Oh, good idea. I'll add.

Reply:
Added two. One to check for the property in the alert-banner component. And a second to the message-types helper that checks the type is a valid option. I added the second one in the helper because if by chance the message types were added onto or changed, I didn't want to duplicate the list of types in the alert banner component file (which I would have had to do for the check).